prow/cmd/deck/job_history.go

```diff
@@ -160,39 +182,63 @@ func readJSON(bucket storageBucket, key string, data interface{}) error {
 }
 
 // Lists the GCS "directory paths" immediately under prefix.
-func (bucket gcsBucket) listSubDirs(prefix string) ([]string, error) {
+func (bucket gcsBucket) listSubDirs(ctx context.Context, prefix string) ([]string, error) {
 	if !strings.HasSuffix(prefix, "/") {
 		prefix += "/"
 	}
 	dirs := []string{}
-	it := bucket.Objects(context.Background(), &storage.Query{
-		Prefix:    prefix,
-		Delimiter: "/",
-	})
+	var it io.ObjectIterator
+	if bucket.storageType == "gs" {
+		it = io.GCSObjectIterator{
+			Iterator: bucket.Objects(ctx, &storage.Query{
+				Prefix:    prefix,
+				Delimiter: "/",
+			}),
+		}
+	} else {
+		var err error
+		it, err = bucket.Opener.Iterator(ctx, fmt.Sprintf("%s://%s/%s", bucket.storageType, bucket.name, prefix), "/")
```

Comment:
Why can we not move the extra-sausage-for-gcs into the `Opener.Iterator` implementation?

Reply:
Fixed